src/pointer.rs

```diff
@@ -906,18 +1222,54 @@ mod tests {
         ];
         for (input, expected) in tests {
             let actual = Pointer::parse(input).map(Pointer::as_str);
-            assert_eq!(
-                actual, expected,
-                "pointer parsing failed to meet expectations
-                \ninput: {input}
-                \nexpected:\n{expected:#?}
-                \nactual:\n{actual:#?}",
-            );
+            assert_eq!(actual, expected);
         }
     }
 
     #[test]
-    fn test_push_pop_back() {
+    fn parse_error_offsets() {
+        let err = Pointer::parse("/foo/invalid~encoding").unwrap_err();
+        assert_eq!(err.pointer_offset(), 4);
+        assert_eq!(err.source_offset(), 8);
+        assert_eq!(err.complete_offset(), 12);
+
+        let err = Pointer::parse("invalid~encoding").unwrap_err();
+        assert_eq!(err.pointer_offset(), 0);
+        assert_eq!(err.source_offset(), 0);
+
+        let err = Pointer::parse("no-leading/slash").unwrap_err();
+        assert!(err.source().is_none());
```

Comment:
Not sure if this should be tested, is adding a source a breaking change? Ditto elsewhere.

Reply:
It was to get coverage on that branch of `source` 😅. I don't know if it'd be considered a breaking change but I would assume so?

Reply:
Created an issue for this one too: https://github.com/chanced/jsonptr/issues/57